hack/test-e2e.sh

```diff
@@ -6,6 +6,13 @@ go clean -testcache
 
 export KAPP_BINARY_PATH="${KAPP_BINARY_PATH:-$PWD/kapp}"
 
+if [ -z "$KAPP_E2E_NAMESPACE" ]; then
```

Comment:
Let's also remove the `-u` flag in set at line number 3, so that we don't get an error when the variable is not already set. We can also set the `pipefail` option similar to the solution in linked PR.

Reply:
Sure!